src/sys/socket/addr.rs

```diff
@@ -688,6 +705,30 @@ impl SockAddr {
         format!("{}", self)
     }
 
+    /// Creates a `SockAddr` struct from libc's sockaddr.
+    pub unsafe fn from_libc_sockaddr(addr: *mut libc::sockaddr) -> Option<SockAddr> {
+        if addr.is_null() {
+            None
+        } else {
+            match AddressFamily::from_i32((*addr).sa_family as i32) {
+                Some(AddressFamily::Unix) => None,
+                Some(AddressFamily::Inet) => Some(SockAddr::Inet(
+                    InetAddr::V4(*(addr as *mut libc::sockaddr_in)))),
+                Some(AddressFamily::Inet6) => Some(SockAddr::Inet(
+                    InetAddr::V6(*(addr as *mut libc::sockaddr_in6)))),
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                Some(AddressFamily::Netlink) => Some(SockAddr::Netlink(
+                    NetlinkAddr(*(addr as *mut libc::sockaddr_nl)))),
+                #[cfg(any(target_os = "ios", target_os = "macos"))]
+                Some(AddressFamily::System) => Some(SockAddr::SysControl(
+                    SysControlAddr(*(addr as *mut sys_control::sockaddr_ctl)))),
+                // No other address families are supported, including netlink and syscontrol.
```

Comment:
Is this a `TODO` or `FIXME` or just impossible? Please clarify this comment further.